tests/infra/clients.py

```diff
@@ -261,13 +265,13 @@ def __init__(
         self.format = format
 
         while connection_timeout >= 0:
+            connection_timeout -= 0.1
             try:
                 self.connect()
                 break
-            except ssl.SSLError:
+            except (ssl.SSLError, ssl.SSLCertVerificationError):
```

Comment:
This exception was not caught here. It is raised when the peer's endorsing certificate does not match the one we expect (here `networkcert.pem`), related to changes implemented by https://github.com/microsoft/CCF/issues/552.